### ISSUE TITLE
MySqlCommandBuilder.DeriveParameters doesn't set DbParameter.Size

### ISSUE THREAD

**Issue by dxmi** [mysql-net/MySqlConnector#1125]
**Software versions**
MySqlConnector version: 2.1.4
Server type and version: MySQL 8.0.12
.NET Framework version: 4.5.2

**Describe the bug**
The MySqlCommandBuilder.DeriveParameters method doesn't set the Size property of DbParameter.

**Code sample**

```csharp
        [Test]
        public void GetStoredProcedureWithBit() {
            var connection = new MySqlConnector.MySqlConnection(dbConnectionString);
            connection.Open();

            using(var command = connection.CreateCommand()) {
                command.CommandText = string.Format("DROP PROCEDURE `{0}`", "procedure_bit");
                try {
                    command.ExecuteNonQuery();
                }
                catch { }
            }
            using(var command = connection.CreateCommand()) {
                command.CommandText =
    @"CREATE DEFINER=`root`@`%` PROCEDURE `procedure_bit`(in param1 bit, in param2 bit(1), out param3 bit(2), out param4 bigint unsigned)
BEGIN
END";
                try {
                    command.ExecuteNonQuery();
                }
                catch { }
            }

            using(var command = connection.CreateCommand()) {
                command.CommandType = CommandType.StoredProcedure;
                command.CommandText = "procedure_bit";

                MySqlConnector.MySqlCommandBuilder.DeriveParameters(command);

                Assert.AreEqual(4, command.Parameters.Count);

                Assert.AreEqual("@param1", command.Parameters[0].ParameterName);
                Assert.AreEqual(DbType.UInt64, command.Parameters[0].DbType);
                Assert.AreEqual(MySqlConnector.MySqlDbType.Bit, command.Parameters[0].MySqlDbType);
                Assert.AreEqual(0, command.Parameters[0].Size);

                Assert.AreEqual("@param2", command.Parameters[1].ParameterName);
                Assert.AreEqual(DbType.UInt64, command.Parameters[1].DbType);
                Assert.AreEqual(MySqlConnector.MySqlDbType.Bit, command.Parameters[1].MySqlDbType);
                Assert.AreEqual(1, command.Parameters[1].Size);

                Assert.AreEqual("@param3", command.Parameters[2].ParameterName);
                Assert.AreEqual(DbType.UInt64, command.Parameters[2].DbType);
                Assert.AreEqual(MySqlConnector.MySqlDbType.Bit, command.Parameters[2].MySqlDbType);
                Assert.AreEqual(2, command.Parameters[2].Size);

                Assert.AreEqual("@param4", command.Parameters[3].ParameterName);
                Assert.AreEqual(DbType.UInt64, command.Parameters[3].DbType);
                Assert.AreEqual(MySqlConnector.MySqlDbType.UInt64, command.Parameters[3].MySqlDbType);
                Assert.AreEqual(0, command.Parameters[3].Size);

            }
        }
```

**Expected behavior**
See asserts in the above test.

**Comment by bgrainger** [comment]
MySqlConnector attaches no significance to the `MySqlParameter.Size` property; i.e., setting it to any value has no effect on the behaviour of the library at all. (Is that a bug? Should it change something?)

Is this bug report/request stemming from a desire for `DeriveParameters` to provide metadata about a stored procedure? `MySqlConnection.GetSchema("Parameters")` might be better way to acquire this information.

**Comment by dxmi** [comment]
Yes, we use `DeiveParameters` to get metadata for procedure parameters.

Thank you for the clue, I tried `MySqlConnection.GetSchema("Parameters")`, but it throws an exception:
```
System.ArgumentException : Value was either too large or too small for an Int32.Couldn't store <4294967295> in CHARACTER_MAXIMUM_LENGTH Column.  Expected type is Int32.
```

It looks like column types are wrong. According to the `reader.GetSchemaTable()`, this column is mapped to Int64.
![MySql_Parameters](https://user-images.githubusercontent.com/37874810/152424414-6965dc31-e52b-4383-8795-cb0212eec6e9.png)

**Comment by bgrainger** [comment]
Fixed in [2.1.6](https://github.com/mysql-net/MySqlConnector/releases/tag/2.1.6).

**Comment by 0xkalawy** [comment]
Does this issue produce [CVE-2024-32655](https://github.com/npgsql/npgsql/security/advisories/GHSA-x9vc-6hfv-hg8c) found in npqsql?
